### PR TITLE
Add initial tests

### PR DESCRIPTION
**Pull request by aljones15**
Adds initial tests for all routes.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -29,17 +29,17 @@ jobs:
...
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
     - run: |
-        npm install
+        npm install --legacy-peer-deps
         cd test
-        npm install
+        npm install --legacy-peer-deps
```
Comment (dlongley) [inline .github/workflows/main.yml]: I think this repo has `.npmrc` with this flag set ... (if it doesn't, it should). In which case, we don't need it in the `ci` scripts.

File: lib/exchangeInstance.js [hunk 2]
```diff
@@ -7,11 +7,11 @@ import {createRequire} from 'node:module';
 import {encodeId, decodeId, generateId} from './helpers.js';
 import {getAppIdentity} from '@bedrock/app-identity';
 import {httpsAgent} from '@bedrock/https-agent';
+import {Ed25519Signature2020} from '@digitalbazaar/ed25519-signature-2020';
+import {ZcapClient} from '@digitalbazaar/ezcap';
```
Comment (dlongley) [inline lib/exchangeInstance.js:11]: Sort imports alphabetically.

File: test/mocha/10-api.js [hunk 3]
```diff
@@ -1,6 +1,204 @@
...
+  shouldHaveInteractService,
+  shouldBeDelegatedZcap
+} from './assertions.js';
+
+const exchange = '/exchanges/:exchangeId';
+const exchangeInstance = '/exchange-instances/:exchangeInstanceId';
+
+describe('API', () => {
+  describe(exchange, () => {
+    it('should get a Vp from a valid service with only one step.', async () => {
```
Comment (dlongley) [inline test/mocha/10-api.js]: I don't think we normally put periods/full stops in test descriptions (unless I'm totally misremembering):

```suggestion
    it('should get a VP from a valid service with only one step', async () => {
```

File: test/mocha/10-api.js [hunk 4]
```diff
@@ -1,6 +1,204 @@
...
+        ...interactResponse,
+        path: interactService.serviceEndpoint,
+        expected: {
+          status: 500
+        }
+      });
+    });
+
+  });
+  describe(`${exchangeInstance}/:stepId`, () => {
```
Comment (dlongley) [inline test/mocha/10-api.js]: Clean up style.

File: test/mocha/assertions.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
+export const shouldNotError = ({response, error, data, path}) => {
```
Comment (dlongley) [inline test/mocha/assertions.js]: We usually use `export function shouldNotError` style here, not arrow functions.

```suggestion
export const shouldNotError = ({response, error, data, path}) => {
```

File: test/mocha/helpers.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
```
Comment (dlongley) [inline test/mocha/helpers.js]:
```suggestion
```

File: test/mock.data.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
```
Comment (dlongley) [inline test/mock.data.js]: No whitespace following copyright header.

```suggestion
```

**Review by dlongley — CHANGES_REQUESTED** [review]
It's always good to have tests ... but we shouldn't put any more effort into these than we already have since we're going to rewrite this module and it's likely to change considerably.

File: lib/exchangeInstance.js (hunk 2 above)
Comment (aljones15) [reply lib/exchangeInstance.js:11]:
```js
import {member} from 'module-name';
```

is that by `member` or by `module-name`?

File: test/mocha/assertions.js (hunk 5 above)
Comment (aljones15) [reply test/mocha/assertions.js]: made the changes from arrows to function here: https://github.com/digitalbazaar/bedrock-vc-exchanger/pull/10/commits/f94cffa35b188259a32394663c082588149327e6

File: test/mocha/10-api.js (hunk 4 above)
Comment (aljones15) [reply test/mocha/10-api.js]: what is the clean up here? Arrow functions in mocha tests? No space between `describe` statements?

File: lib/exchangeInstance.js (hunk 2 above)
Comment (dlongley) [reply lib/exchangeInstance.js:11]: It's by member.

File: test/mocha/10-api.js (hunk 4 above)
Comment (dlongley) [reply test/mocha/10-api.js]: There's random whitespace to remove. And make sure if there's no whitespace before `describe` that's consistent throughout the file.

File: .github/workflows/main.yml (hunk 1 above)
Comment (aljones15) [reply .github/workflows/main.yml]: Dropped the `--legacy-peer-deps` flag here: https://github.com/digitalbazaar/bedrock-vc-exchanger/pull/10/commits/bce4d8a4f67e66fd6dc9d427aa9d408d24d5dbe6

File: test/mocha/10-api.js (hunk 4 above)
Comment (aljones15) [reply test/mocha/10-api.js]: Addressed here: https://github.com/digitalbazaar/bedrock-vc-exchanger/pull/10/commits/420027fb6d9a1bdaf1ebb8acdad8d953f56e260d

File: lib/exchangeInstance.js (hunk 2 above)
Comment (aljones15) [reply lib/exchangeInstance.js:11]: I used the newer listing rules to alphabetize them: https://github.com/digitalbazaar/bedrock-vc-exchanger/pull/10/commits/af871c95b640584170d0a89b8179c1c3d557c090

File: test/mocha/10-api.js [hunk 8]
```diff
@@ -1,6 +1,203 @@
...
+      const expected = {status: 500};
+      shouldError({response, error, data, path, expected});
+    });
+    it('should error if service doesn\'t have an initial step.', async () => {
+      const path = 'exchanges/noInitialStep';
+      const {response, error, data} = await api.post({path});
+      const expected = {status: 500};
+      shouldError({response, error, data, path, expected});
+    });
+    it('should error if in interact a service has an invalid type.',
```
Comment (dlongley) [inline test/mocha/10-api.js]: These test descriptions all still have periods at the end -- could you please remove to keep with consistent styling?

File: test/mocha/assertions.js [hunk 9]
```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
```
Comment (dlongley) [inline test/mocha/assertions.js]:
```suggestion
```

File: test/mocha/helpers.js [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import {Agent} from 'node:https';
+import {decodeSecretKeySeed} from 'bnid';
+import {Ed25519Signature2020} from '@digitalbazaar/ed25519-signature-2020';
+import {httpClient} from '@digitalbazaar/http-client';
+
+const agent = new Agent({rejectUnauthorized: false});
+const baseUrl = 'https://localhost:18443';
+const didKeyDriver = didKey.driver();
+
+const makeRequest = async ({url, json, headers, method}) => {
```
Comment (dlongley) [inline test/mocha/helpers.js]: Helpers should be at the bottom of the file. Ideally prefixed with `_`. Use `function` style instead of arrow.

**Review by dlongley — APPROVED** [review]
Thanks! A few more nits.

**Comment by dlongley** [comment]
@gannan08 you may want to look more closely at the tests than I did to ensure they are testing the right things.

File: test/mocha/helpers.js (hunk 10 above)
Comment (aljones15) [reply test/mocha/helpers.js]: addressed here: https://github.com/digitalbazaar/bedrock-vc-exchanger/pull/10/commits/ddca903f15d3e9961a2b50a89ca4d3fd7ac6e1e8

File: test/mocha/10-api.js (hunk 8 above)
Comment (aljones15) [reply test/mocha/10-api.js]: addressed here: https://github.com/digitalbazaar/bedrock-vc-exchanger/pull/10/commits/ddca903f15d3e9961a2b50a89ca4d3fd7ac6e1e8

**Review by gannan08 — APPROVED** [review]
LGTM. Post-merge, we will need to test conformance statements found in the specification.

**Pull request merged**